From d5af9bbfbc683461f19b3357aa90aeefa800fd76 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Wed, 29 Aug 2018 18:00:19 +0900
Subject: [PATCH] arm64: dts: r8a7795-h3ulcb: Fix source clock for DU2

This patch changes the clock for DU2, supply the initial value of
versaclock directly to the DU2.

The initial value of versaclock out4 is 33 MHz.

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Signed-off-by: Yusuke Goda <yusuke.goda.sx@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts
index 4f15df0..45f64d8 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-h3ulcb.dts
@@ -84,6 +84,13 @@
 		/* Initial value of versaclock out3 */
 		clock-frequency = <33000000>;
 	};
+
+	versaclock5_out4: versaclk-4 {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		/* Initial value of versaclock out4 */
+		clock-frequency = <33000000>;
+	};
 };
 
 &du {
@@ -94,7 +101,7 @@
 		 <&cpg CPG_MOD 727>,
 		 <&versaclock5 1>,
 		 <&versaclock5_out3>,
-		 <&versaclock5 4>,
+		 <&versaclock5_out4>,
 		 <&versaclock5 2>;
 	clock-names = "du.0", "du.1", "du.2", "du.3", "lvds.0",
 		      "dclkin.0", "dclkin.1", "dclkin.2", "dclkin.3";
-- 
2.7.4

